### ISSUE TITLE
Conflicting rules for opening brace

### ISSUE THREAD

**Issue by Craige** [escapestudios/Symfony2-coding-standard#12]
The standard seems to not know what to do when it encounters a multiline function declaration such as the following

        public function __construct(
            $content = '',
            $status = 200,
            $headers = array()
        ) {
            //...
        }
PHPCS reports

    22 | ERROR | [x] Opening brace should be on a new line

If I put it on a new line...

        public function __construct(
            $content = '',
            $status = 200,
            $headers = array()
        )
        {
            //...
        }
 
I get the following:

     22 | ERROR | [x] There must be a single space between the closing
        |       |     parenthesis and the opening brace of a multi-line
        |       |     function declaration; found newline

**Comment by wickedOne** [comment]
according to the [psr-2](http://www.php-fig.org/psr/psr-2/) standard your first example is correct and the ``Opening brace should be on a new line`` should not be triggered. this is actually caused by the generic [FunctionDeclarationSniff](https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/PEAR/Sniffs/Functions/FunctionDeclarationSniff.php#L375) which lacks a check for a multi-line function declaration. you might want to consider opening an issue [over here](https://github.com/squizlabs/PHP_CodeSniffer/issues)

**Comment by djoos** [comment]
@Craige: do give us a shout if you're going to open a PR in the squizlabs/PHP_CodeSniffer-repo.

In the meantime I'll close this issue over here.

Thanks guys!

Kind regards,
David

**Comment by gl3n** [comment]
I encounter the same issue ... very annoying !

@Craige is there any news about this issue ?

**Comment by fre5h** [comment]
I also always get this issue. Hope it will be fixed at some time.

**Comment by djoos** [comment]
Reopened!

**Comment by djoos** [comment]
Anyone who would like to open an issue in the squizlabs/PHP_CodeSniffer-repo, so we can close this one off here?

Thanks in advance!

Kind regards,
David

**Comment by nesl247** [comment]
I fixed this is my company's ruleset that rides atop of this one with:

```
<rule ref="Generic.Functions.OpeningFunctionBraceBsdAllman.BraceOnSameLine">
    <severity>0</severity>
</rule>
```

Maybe it isn't the proper fix, but it worked for us for now.

**Comment by gl3n** [comment]
Hi @djoos 
Did you see the response on the issue I created on PHP_CodeSniffer repo ?
https://github.com/squizlabs/PHP_CodeSniffer/issues/661

**Comment by nesl247** [comment]
@gl3n After looking at it further, they are right. The ruleset in this repository is the one causing the issue. We are importing the Zend ruleset and not excluding `Generic.Functions.OpeningFunctionBraceBsdAllman.BraceOnSameLine` like I mentioned above.

**Comment by gl3n** [comment]
Hi @nesl247 could you do a pull request on the repo with this fix ?

**Comment by djoos** [comment]
Thanks for your help in getting to the bottom of this guys!

Kind regards,
David

**Comment by joan16v** [comment]
I have the same issue, is it solved?

**Comment by zquintana** [comment]
I'm getting the same issue, except for me now it doesn't detect fail when an open brace isn't on a new line at all.

E.g this passes
``` php
public function test(){
    return null;
}
```
